os/kernel/init/os_start.c

```diff
@@ -317,33 +319,46 @@ const struct tasklist_s g_tasklisttable[NUM_TASK_STATES] =
 		&g_waitingforfin,
 		TLIST_ATTR_PRIORITIZED
 	},
+#ifndef CONFIG_DISABLE_SIGNALS
 	{                                              /* TSTATE_WAIT_SIG */
 		&g_waitingforsignal,
 		0
-	}
+	},
+#else
+	{                                             
+		NULL,
+		0
+	},
+#endif
 #ifndef CONFIG_DISABLE_MQUEUE
-	,
 	{                                              /* TSTATE_WAIT_MQNOTEMPTY */
 		&g_waitingformqnotempty,
 		TLIST_ATTR_PRIORITIZED
 	},
 	{                                              /* TSTATE_WAIT_MQNOTFULL */
 		&g_waitingformqnotfull,
 		TLIST_ATTR_PRIORITIZED
+	},
+#else
+	{                                             
+		NULL,
+		0
+	},
+	{                                             
+		NULL,
+		0
 	}
 #endif
 #ifdef CONFIG_PAGING
 	,
 	{                                              /* TSTATE_WAIT_PAGEFILL */
 		&g_waitingforfill,
 		TLIST_ATTR_PRIORITIZED
-	}
-#endif
-#ifdef CONFIG_SIG_SIGSTOP_ACTION
-	,
-	{                                              /* TSTATE_TASK_STOPPED */
-		&g_stoppedtasks,
-		0                                            /* See tcb->prev_state */
```

Comment:
This is removed. Is this intended?

Reply:
We dont have an enum or list for it. So, we removed it.